src/GitTfs.VsCommon/TfsHelper.Common.cs

```diff
@@ -667,24 +661,35 @@ protected ITfsChangeset BuildTfsChangeset(Changeset changeset, IGitTfsRemote rem
             return tfsChangeset;
         }
 
-        protected virtual bool HasWorkItems(Changeset changeset)
+        protected virtual IEnumerable<TfsWorkitem> GetWorkItems(Changeset changeset)
         {
-            // This method wraps changeset.WorkItems, because
-            // changeset.WorkItems might result to ConnectionException: TF26175: Team Foundation Core Services attribute 'AttachmentServerUrl' not found.
-            // in this case assume that it is initialized to null
-            // NB: in VS2011 a new property appeared (AssociatedWorkItems), which works correctly
-            WorkItem[] result = null;
+            var workItems = changeset.AssociatedWorkItems.Select(wi => new TfsWorkitem
+            {
+                Id = wi.Id, 
+                Title = wi.Title
+            }).ToList();
+
             try
             {
-                result = Retry.Do(() => changeset.WorkItems);
+                var updatedWorkItems = changeset.WorkItems.Join(workItems, wi => wi.Id, twi => twi.Id, (wi, twi) =>
+                {
+                    twi.Description = wi.Description;
+                    twi.Url = Linking.GetArtifactUrl(wi.Uri.AbsoluteUri);
+                    return twi;
+                });
+
+                return updatedWorkItems;
             }
-            catch (ConnectionException exception)
+            catch (WorkItemTypeDeniedOrNotExistException e)
             {
-                if (!exception.Message.StartsWith("TF26175:"))
-                    throw;
+                Trace.TraceWarning(e.Message);
+                return workItems;
```

Comment:
Here we return the only half filled `TfsWorkitem`s in case we have an error.
Furthermore, if e.g.  AssociatedWorkItems.Lenght == 3 and we fail on the first work item when joining,
then all the subsequent work items will be silently ignored. I believe your first version didn't have this problem.

On the other hand you mentioned that due the prefiltering on the AssociatedWorkItems the error no longer exists.
Perhaps this is an artificial problem only?
Can you double check if this is true? If so, we So if there are e.g.